nomad/core_sched.go

```diff
@@ -823,7 +827,7 @@ func collectClaimsToGCImpl(vol *structs.CSIVolume) ([]gcClaimRequest, map[string
 				continue
 			}
 			nodeClaims[alloc.NodeID]++
-			if alloc.Terminated() {
+			if runningAllocs || alloc.Terminated() {
```

Comment:
I don't love just parameterizing this (or forcing an claim release on an alloc that's running), but I think it's the way that job stop works right now. There is probably a better fix for this that involves actually terminating all the allocs first.

Reply:
Yeah it's not a beautiful API. But it's also not public so we can revisit it after shipping. I think there's going to be some performance optimization we want to do of the whole claim release path post-RC anyways (I'm trying to put together an issue to describe how I want to test that out).